tests/framework/extensions/clusters/clusters.go

```diff
@@ -625,16 +626,29 @@ func AgentAffinityConfigHelper(advancedClusterAffinity *management.Affinity) *co
 func HardenK3SRKE2ClusterConfig(clusterName, namespace string, clustersConfig *ClusterConfig, machinePools []apisV1.RKEMachinePool, cloudCredentialSecretName string) *apisV1.Cluster {
```

Comment:
nit; the outcome is different for k3s vs. rke2. Wouldn't it make more sense to have 2 separate functions in this case? One for k3s, one for rke2

Reply:
I fiddled around with it and broke it up and it does look a bit better having it separated. Thinking further on it, in the future, it probably will be easier to maintain/add onto these tests with it separated as well.